pkg/registry/core/pod/storage/eviction.go

```diff
@@ -138,46 +138,54 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		return &metav1.Status{
 			Status: metav1.StatusSuccess}, nil
 	}
-	var rtStatus *metav1.Status
 	var pdbName string
 	err = func() error {
 		pdbs, err := r.getPodDisruptionBudgets(ctx, pod)
 		if err != nil {
 			return err
 		}
 
-		if len(pdbs) > 1 {
-			rtStatus = &metav1.Status{
-				Status:  metav1.StatusFailure,
-				Message: "This pod has more than one PodDisruptionBudget, which the eviction subresource does not support.",
-				Code:    500,
-			}
-			return nil
-		}
 		if len(pdbs) == 0 {
 			return nil
 		}
 
-		pdb := &pdbs[0]
-		pdbName = pdb.Name
 		refresh := false
-		err = retry.RetryOnConflict(EvictionsRetry, func() error {
-			if refresh {
-				pdb, err = r.podDisruptionBudgetClient.PodDisruptionBudgets(pod.Namespace).Get(pdbName, metav1.GetOptions{})
+		for _, withDryRun := range []bool{true, false} {
+			var seenErr error
+			for i := range pdbs {
+				err = retry.RetryOnConflict(EvictionsRetry, func() error {
+					pdbName = pdbs[i].Name
+					var pdb *policyv1beta1.PodDisruptionBudget
+					if refresh {
+						pdb, err = r.podDisruptionBudgetClient.PodDisruptionBudgets(pod.Namespace).Get(pdbName, metav1.GetOptions{})
+						if err != nil {
+							return err
+						}
+					} else {
+						pdb = &pdbs[i]
+					}
+					// Try to verify-and-decrement
+
+					// If it was false already, or if it becomes false during the course of our retries,
+					// raise an error marked as a 429.
+					if err = r.checkAndDecrement(pod.Namespace, pod.Name, *pdb, withDryRun); err != nil {
+						refresh = true
+						return err
+					}
+					return nil
+				})
 				if err != nil {
-					return err
+					seenErr = err
```

Comment:
In which scenarios do we want to keep going if we get an error here? Most errors that could be returned from the checkAndDecrement function means that eviction of the pod is blocked by a pdb. In this case, it doesn't seem like we should keep trying to update the remaining pdbs.